torch/csrc/cudnn/Conv.cpp

```diff
@@ -225,6 +225,14 @@ void cudnn_convolution_forward(
   size_t workspaceSize;
   CHECK(cudnnGetConvolutionForwardWorkspaceSize(handle, idesc.desc, wdesc.desc,
       cdesc.desc, odesc.desc, fwdAlg, &workspaceSize));
+  size_t memFree;
+  size_t memTotal;
+  cudaMemGetInfo(&memFree, &memTotal);
```

Comment:
It's a good point, yeah we should. There are some cases where free memory reported by the driver will be nearly 0, but the caching allocator will have plenty of free blocks to use.